Replace pdflatex with LuaLatex and latexmk

PatrickBaus opened the pull request:
Thank you so much for making this software open source. I am still learning to use it and I am currently working on improving the LaTex backend to finally integrate Facture-X/Zugferd and Order-X into the billing backend.

I would  like to push those changes upstream as I progress if you are interested.

This PR starts with a small change to the LaTex backend replacing `pdflatex` with `lualatex` and `latexmk`. The latter has the following advantages over calling plain `pdflatex` or `lualatex`:

- `latexmk` will call tex multiple times if several passes are required by the tex engine.
- `latexmk` has the `-cd` flag which enters the temp directory during compilation and allows relative imports of auxiliary files like images (e.g. company logos or scanned signatures)

I chose to replace `pdflatex` with `lualatex` because `lualatex` offers a more modern backend with better utf-8 support. It also represent the current development direction instead of `pdflatex` which will see no new features added. This change will make generating compliant PDFs for Facture-X a lot easier later on.

The change from `lualatex` to  `pdflatex` changes the default Computer Modern font (or rather European Computer Modern when loading the T1 font using `fontenc`) used by  `pdflatex` to Latin Modern as this is a unicode compliant default font. Since the LaTex templates are demo templates I do not see an issue here.

genglert commented:
Thx for the PR. We'll discuss internally if it's better to get your current modifications, or if it's better to get several Latex backends, then I'll make a complete feedback.

PatrickBaus commented:
@genglert Sure, do let me know. I can also rework the PR a bit if needed.

genglert reviewed (changes requested):
Hi ! Sorry for the delay we were very busy.
- could you squash your commits into one (I prefer to regroup code/tests/comments if possible).
- could you add an item in the CHANGELOG.txt (user side/apps/billing) to indicate that Latex must have 'latexmk' & 'lualatex' (and so install the package 'latexmk' on Ubuntu for example).
Thx

Edit: you can also add you as author in the header of "creme/billing/exporters/latex.py"

PatrickBaus commented:
@genglert Thank you for getting back on this. I have squashed the commits into one commit and added the changes to the commit message. I have also updated the `CHANGELOG.txt` file.

I added a copyright line to the `creme/billing/exporters/latex.py` file. If this is not what you meant with your comment do feel free to edit or remove it.

My next step will be to add some code to copy additional files from the template directory into the tmp folder used for compilation. This allows to use logos and other images in the LaTex files.

genglert commented:
> My next step will be to add some code to copy additional files from the template directory into the tmp folder used for compilation. This allows to use logos and other images in the LaTex files.

Could you open a thread in the "Discussions" tool on GitHub to discuss about that (spoiler: there are already solutions/workarounds for that without additional need of code + improving the code could be better done without file copy) ?
